ioeffect/src/main/scala/scalaz/ioeffect/MonadIOInstances.scala

```diff
@@ -0,0 +1,94 @@
+package scalaz.ioeffect
+
+import scalaz._
+import Scalaz._
+
+sealed abstract class MonadIOInstances extends MonadIOInstances1 {
+
+  implicit val taskMonadIO: MonadIO[Task, Throwable] = new MonadIO[Task, Throwable] {
+    override def liftIO[A](io: IO[Throwable, A])(implicit M: Monad[Task]): Task[A] = io
+  }
+}
+
+sealed abstract class MonadIOInstances1 extends MonadIOInstances2 {
+
+  implicit def ioMonadIO[E]: MonadIO[IO[E, ?], E] = new MonadIO[IO[E, ?], E] {
+    override def liftIO[A](io: IO[E, A])(implicit M: Monad[IO[E, ?]]): IO[E, A] = io
+  }
+}
+
+sealed abstract class MonadIOInstances2 {
+  import Isomorphism._
+
+  def fromIso[F[_], G[_], E](D: F <~> G)(implicit E: MonadIO[G, E], M1: Monad[G]): MonadIO[F, E] =
+    new IsomorphismMonadIO[F, G, E] {
+      override def G: MonadIO[G, E] = E
+      override def M: Monad[G]      = M1
+      override def iso: F <~> G     = D
+    }
+
+  implicit def identityTMonadIO[M[_], E](implicit M: MonadIO[M, E], M1: Monad[M]): MonadIO[IdT[M, ?], E] =
```

Comment:
I don't think these need to go on the lowest priority trait. You can put them into the companion.

Reply:
👍